src/bytes.rs

```diff
@@ -58,27 +59,42 @@ const _: () = {
 
 extern "C" fn noop(_: *const (), _: usize) {}
 
+const IS_LITTLE_ENDIAN: bool = unsafe { mem::transmute::<u16, [bool; 2]>(1)[0] };
```

Comment:
I thought about this initially, but I don't know the "endianness of `const`-eval" 😬 

Since [`::byteorder` uses a `cfg`](https://github.com/BurntSushi/byteorder/blob/18f32ca3a41c9823138e782752bc439e99ef7ec8/src/lib.rs#L1842C3-L1842C32), I think we should be doing the same:

```suggestion
const IS_LITTLE_ENDIAN: bool = cfg!(target_endian = "little");
```

Reply:
that's definitely much cleaner, I had assumed from your snippet that the `cfg` wasn't around or something xD